lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -334,6 +334,8 @@ def create(self, trans, payload, **kwd):
         :rtype:     dict
         :returns:   element view of new history
         """
+        if not trans.galaxy_session:
+            raise exceptions.MessageException("Only users with a session can create histories.")
```

Comment:
I think this could deserve its own exception class?

Reply:
I created a specific exception here https://github.com/galaxyproject/galaxy/pull/11846/commits/4460b0ce3476d6174c83b4026f5292bfa8b0eb14 (sorry for the auto-formatting)